community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/phases/DeprecationWarnings.scala

```diff
@@ -75,8 +75,8 @@ object ProcedureWarnings extends VisitorPhase {
 
   private def findWarnings(statement: Statement): Set[InternalNotification] =
     statement.treeFold(Set.empty[InternalNotification]) {
-      case f@ResolvedCall(ProcedureSignature(name, _, _, _, _, _, true), _, _, _, _) =>
-        (seq) => (seq + ProcedureWarningNotification(f.position, name.toString), None)
+      case f@ResolvedCall(ProcedureSignature(name, _, _, _, _, _, Some(warning)), _, _, _, _) =>
```

Comment:
There's no benefit from using lists currently, if and when it's something we want/need it's easy to fix then.